OneCommunityGlobal/HGNRest
# Demi_fix_duplicate_id_error

# Description
Fixes Phase 2 WBS Bugs line item 3 “Material List” 
Material items and purchase record were not having a unique ID created causing an error in the console.

## Related PRS (if any):
[PR 1813
](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1813)
## Main changes explained:
- Delete code that prevented a unique ID from being made in both purchase records and update records
- Deleted existing records in MongoDB to prevent an error from occurring since the existing records lacked an ID
- Created new records with unique ID's attached

## How to test:

1. git pull
2. check into current branch
3. npm install
4. npm run build && npm start to run this PR locally
5. Clear site data/cache in local storage
6. log in as any user
7. go to http://localhost:3000/bmdashboard/login to confirm your login credentials
8. go to http://localhost:3000/bmdashboard/materials
9. select "View" under "Purchases"
10. Verify that you do not receive a console error for duplicate keys
11. select "View" under "Updates"
12. Verify that you do not receive a console error for duplicate keys

## Screenshots or videos of changes:
<img width="1147" alt="Screenshot 2024-01-04 at 1 17 33 PM (2)" src="https://github.com/OneCommunityGlobal/HGNRest/assets/93569791/e502057b-56dd-4080-ae9d-41af97b0de31">

## Note:
Use dev branch to test front end functionality.
